Compilation issues on Windows 11 ARM64 (macOS host)

I get the following output when opening the project in VS Code. Note that `cargo build` in powershell works fine. Do you have any idea how to fix this?

```
Caused by:
  process didn't exit successfully: `C:\Users\pronebird\Nym\nym-vpn-client\nym-vpn-core\target\debug\build\nym-gateway-probe-0ecbed34c3f46f9b\build-script-build` (exit code: 101)
  --- stdout
  cargo:rerun-if-env-changed=TARGET
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64-pc-windows-msvc
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64_pc_windows_msvc
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS
  cargo:rerun-if-changed=C:\Users\pronebird\Nym\nym-vpn-client\nym-vpn-core\target\debug\build\nym-gateway-probe-d9c50f4d2ec51434\out\libgo.h

  --- stderr
  # github.com/nymtech/nym-vpn-client/nym-vpn-core/crates/nym-gateway-probe/netstack_ping
  In file included from _cgo_export.c:4:
  gen.go:51:13: warning: inline function 'NetstackCall_ping_cb' declared weak [-Wattributes]
  # github.com/nymtech/nym-vpn-client/nym-vpn-core/crates/nym-gateway-probe/netstack_ping
  .\gen.go:51:13: warning: inline function 'NetstackCall_ping_cb' declared weak [-Wattributes]
     51 | inline void NetstackCall_ping_cb(const void *f_ptr, struct NetstackResponseRef resp, const void *slot) {
        |             ^~~~~~~~~~~~~~~~~~~~

  thread 'main' panicked at C:\Users\pronebird\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rust2go-0.3.17\src\build.rs:168:14:
  Unable to generate bindings: NotExist("C:\\Users\\pronebird\\Nym\\nym-vpn-client\\nym-vpn-core\\target\\debug\\build\\nym-gateway-probe-d9c50f4d2ec51434\\out\\libgo.h")
  stack backtrace:
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


2025-04-07T14:46:20.4508835+02:00 ERROR FetchBuildDataError: error: failed to run custom build command for `nym-gateway-probe v1.7.0-beta (C:\Users\pronebird\Nym\nym-vpn-client\nym-vpn-core\crates\nym-gateway-probe)`
note: To improve backtraces for build dependencies, set the CARGO_PROFILE_DEV_BUILD_OVERRIDE_DEBUG=true environment variable to enable debug information generation.

Caused by:
  process didn't exit successfully: `C:\Users\pronebird\Nym\nym-vpn-client\nym-vpn-core\target\debug\build\nym-gateway-probe-0ecbed34c3f46f9b\build-script-build` (exit code: 101)
  --- stdout
  cargo:rerun-if-env-changed=TARGET
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64-pc-windows-msvc
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64_pc_windows_msvc
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS
  cargo:rerun-if-changed=C:\Users\pronebird\Nym\nym-vpn-client\nym-vpn-core\target\debug\build\nym-gateway-probe-d9c50f4d2ec51434\out\libgo.h

  --- stderr
  # github.com/nymtech/nym-vpn-client/nym-vpn-core/crates/nym-gateway-probe/netstack_ping
  In file included from _cgo_export.c:4:
  gen.go:51:13: warning: inline function 'NetstackCall_ping_cb' declared weak [-Wattributes]
  # github.com/nymtech/nym-vpn-client/nym-vpn-core/crates/nym-gateway-probe/netstack_ping
  .\gen.go:51:13: warning: inline function 'NetstackCall_ping_cb' declared weak [-Wattributes]
     51 | inline void NetstackCall_ping_cb(const void *f_ptr, struct NetstackResponseRef resp, const void *slot) {
        |             ^~~~~~~~~~~~~~~~~~~~

  thread 'main' panicked at C:\Users\pronebird\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rust2go-0.3.17\src\build.rs:168:14:
  Unable to generate bindings: NotExist("C:\\Users\\pronebird\\Nym\\nym-vpn-client\\nym-vpn-core\\target\\debug\\build\\nym-gateway-probe-d9c50f4d2ec51434\\out\\libgo.h")
  stack backtrace:
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


2025-04-07T14:47:39.9774029+02:00 ERROR FetchBuildDataError: error: failed to run custom build command for `nym-gateway-probe v1.7.0-beta (C:\Users\pronebird\Nym\nym-vpn-client\nym-vpn-core\crates\nym-gateway-probe)`
note: To improve backtraces for build dependencies, set the CARGO_PROFILE_DEV_BUILD_OVERRIDE_DEBUG=true environment variable to enable debug information generation.

Caused by:
  process didn't exit successfully: `C:\Users\pronebird\Nym\nym-vpn-client\nym-vpn-core\target\debug\build\nym-gateway-probe-0ecbed34c3f46f9b\build-script-build` (exit code: 101)
  --- stdout
  cargo:rerun-if-env-changed=TARGET
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64-pc-windows-msvc
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS_aarch64_pc_windows_msvc
  cargo:rerun-if-env-changed=BINDGEN_EXTRA_CLANG_ARGS
  cargo:rerun-if-changed=C:\Users\pronebird\Nym\nym-vpn-client\nym-vpn-core\target\debug\build\nym-gateway-probe-d9c50f4d2ec51434\out\libgo.h

  --- stderr
  # github.com/nymtech/nym-vpn-client/nym-vpn-core/crates/nym-gateway-probe/netstack_ping
  In file included from _cgo_export.c:4:
  gen.go:51:13: warning: inline function 'NetstackCall_ping_cb' declared weak [-Wattributes]
  # github.com/nymtech/nym-vpn-client/nym-vpn-core/crates/nym-gateway-probe/netstack_ping
  .\gen.go:51:13: warning: inline function 'NetstackCall_ping_cb' declared weak [-Wattributes]
     51 | inline void NetstackCall_ping_cb(const void *f_ptr, struct NetstackResponseRef resp, const void *slot) {
        |             ^~~~~~~~~~~~~~~~~~~~

  thread 'main' panicked at C:\Users\pronebird\.cargo\registry\src\index.crates.io-1949cf8c6b5b557f\rust2go-0.3.17\src\build.rs:168:14:
  Unable to generate bindings: NotExist("C:\\Users\\pronebird\\Nym\\nym-vpn-client\\nym-vpn-core\\target\\debug\\build\\nym-gateway-probe-d9c50f4d2ec51434\\out\\libgo.h")
  stack backtrace:
  note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

--- a/rust2go/src/build.rs
+++ b/rust2go/src/build.rs
@@ -9,6 +9,13 @@ use std::{
 
 use rust2go_cli::Args;
 
+/// Static lib extension on non-Windows platforms.
+#[cfg(not(windows))]
+const LIB_EXT: &str = ".a";
+/// Static lib extension on Windows.
+#[cfg(windows)]
+const LIB_EXT: &str = ".lib";
+
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub enum LinkType {
     Static,
@@ -149,6 +156,8 @@ pub trait GoCompiler {
     fn go_build(&self, go_src: &Path, link: LinkType, output: &Path);
 
     fn build(&self, go_src: &Path, binding_name: &str, link: LinkType, copy_lib: &CopyLib) {
+        use std::env::consts::DLL_PREFIX;
+
         let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
         let out_filename = filename(link);
         let output = out_dir.join(&out_filename);
@@ -191,7 +200,7 @@ pub trait GoCompiler {
         }
 
         let bindings = bindgen::Builder::default()
-            .header(out_dir.join("libgo.h").to_str().unwrap())
+            .header(out_dir.join(format!("{DLL_PREFIX}go.h")).to_str().unwrap())
             .parse_callbacks(Box::new(bindgen::CargoCallbacks::new()))
             .generate()
             .expect("Unable to generate bindings");
@@ -326,7 +335,7 @@ fn filename(link_type: LinkType) -> String {
     use std::env::consts::{DLL_PREFIX, DLL_SUFFIX};
 
     match link_type {
-        LinkType::Static => format!("{DLL_PREFIX}go.a"),
+        LinkType::Static => format!("{DLL_PREFIX}go{LIB_EXT}"),
         LinkType::Dynamic => format!("{DLL_PREFIX}go{DLL_SUFFIX}"),
     }
 }

